include/s3select_json_parser.h

```diff
@@ -369,13 +373,16 @@ void dec_key()
   }
 }
 
-void new_value(s3selectEngine::value& v,size_t json_index)
+void new_value(s3selectEngine::value& v,size_t json_index, json_key_value_t key_path_value, bool json_format = false)
```

Comment:
`json_key_value_t` is "heavy".
using passing argument `by value` means that it is re-construct on runtime, and that could result in higher consumption of CPU.
it is better to use a reference for that parameter.